### ISSUE TITLE
Android fails to compile with AGP 7.3.0-beta03

### ISSUE THREAD

**Issue by headsvk** [icerockdev/moko-resources#353]
Hi! I tried updating Android Gradle Plugin to `7.3.0-beta03` and Moko Resource to `0.20.1` and I can't compile Android anymore.

Steps:
1. Install [Android Studio Dolphin Beta 3](https://developer.android.com/studio/preview)
2. Update Android Gradle Plugin in this project to `7.3.0-beta03` [as in this fork](https://github.com/headsvk/moko-resources/pull/1/files)
3. Run `sample.android-mpp-app`

Gradle fails with:
```
Task :sample:android-mpp-app:compileDebugKotlinAndroid FAILED
e: ..\moko-resources\sample\android-mpp-app\build\generated\moko\androidMain\src\com\icerockdev\library\MR.kt: (16, 77): Unresolved reference: common_name
```

**Comment by headsvk** [comment]
I investigated this and it looks like AGP 7.3.0 is ignoring resources in the `generated` folder. It might be a bug in AGP but I don't know enough to create a ticket there.

**Comment by Alex009** [comment]
hi @headsvk 
as i see android gradle plugin read resources dirs before our generation dir was added. why - i not find yet. i think we should wait release version of AGP before own changes.
for now you can add this to build.gradle:
```
android {
    sourceSets.getByName("main").res.srcDir(File(buildDir, "generated/moko/androidMain/res"))
}
```

**Comment by YektaDev** [comment]
I had a somehow similar issue. For me, assets were missing in builds; at first I blamed R8 aggressive mode, but it turned out to be this issue. I think it's safer to use this in general, until there's an update:
```kotlin
android {
    sourceSets["main"].apply {
        assets.srcDir(File(buildDir, "generated/moko/androidMain/assets"))
        res.srcDir(File(buildDir, "generated/moko/androidMain/res"))
    }
}
```

**Comment by ln-12** [comment]
Hey @Alex009, is your suggested fix still the way to go or could you investigate the problem further? AGP 7.3.0 is stable now.

**Comment by chris-hatton** [comment]
@sproctor In my case Moko is applied to an intermediate (hierarchical) source-set, and in this scenario even with AGP `8.0.0-alpha08` the problem persists. However [the workaround](AGP `8.0.0-alpha08`) still works 👍

**Comment by Alex009** [comment]
will be released in 0.21.0
